usb/esp_tinyusb/include/tusb_msc_storage.h

```diff
@@ -0,0 +1,166 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stddef.h>
+#include "esp_err.h"
+#include "driver/sdmmc_host.h"
+
+/* structs, unions
+   ********************************************************************* */
+
+/**
+ * @brief enum for storage media that are supported
+ */
+typedef enum {
+    TINYUSB_MSC_STORAGE_INVALID,
+    TINYUSB_MSC_STORAGE_SPI,
+    TINYUSB_MSC_STORAGE_SDMMC,
+    TINYUSB_MSC_STORAGE_MAX,
+} tinyusb_msc_storage_type_t;
+
+#if SOC_SDMMC_HOST_SUPPORTED
+/**
+ * @brief Configuration structure for sdmmc initialization
+ *
+ * User configurable parameters that are used while
+ * initializing the sdmmc media.
+ */
+typedef struct {
+    sdmmc_host_t host;
+    sdmmc_slot_config_t slot_config;
+} tinyusb_msc_sdmmc_config_t;
+#endif
+
+/**
+ * @brief Configuration structure for spiffs initialization
+ *
+ * User configurable parameters that are used while
+ * initializing the SPI Flash media.
+ */
+typedef struct {
+    // Place holder. Currently, nothing to configure
+} tinyusb_msc_spiffs_config_t;
+
+/**
+ * @brief Configuration structure for storage initialization
+ */
+typedef struct {
+    tinyusb_msc_storage_type_t storage_type;
+    union {
```

Comment:
As mentioned above, recommend having two separate functions for initializing MSC with SPI flash and SDMMC, in that case each can have its own config structure.

(Not only this allows the program to dynamically choose which of the implementations to use, it also allows the linker to remove unused code if one of the implementations is not used.)

Reply:
This is done in different way.
Instead of two different APIs, I am using a parameter `const tinyusb_msc_storage_type_t storage_type` to differentiate between two types.
The type of media is seleted by user at application (example) level using menuconfig. It is SPIFLASH by default.

Reply:
`tinyusb_msc_storage_type_t storage_type;` is not required anymore.